src/internal_modules/roc_rtcp/session.h

```diff
@@ -44,7 +44,7 @@ class Session {
 
     //! Parse and process incoming packet.
     //! Invokes session hooks methods during processing.
-    void process_packet(const packet::PacketPtr& packet);
+    virtual ROC_ATTR_NODISCARD status::StatusCode write(const packet::PacketPtr& packet);
```

Comment:
I'd keep it `status::StatusCode process_packet()` and don't convert rtcp::Session to packet writer.

The semantics is quite different. Packets writers are chained to form sender pipeline that handles transport packets. rtpc::Session is not part of that pipeline. It handles control packets, it's not chained. Instead it invokes various hooks bases on what it found in the packet being processed.

I think it makes sense to keep its interface distinguishable from other writers.

Reply:
BTW, thanks for noticing that we need statuses in rtcp::Session, I didn't think about it.

Later we'll need to forward these errors too. I've added it to #614.

Reply:
I think also about packet::IComposer, probably it's worth to replace `bool` via `status::StatusCode` too.